### PR TITLE
Add rocksdb support

### PR DESCRIPTION
**Pull request by houjun**
Tested with rocksdb 8.1.1
Also added a serial kvtag test

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -459,6 +459,14 @@ if(PDC_ENABLE_FASTBIT)
     set(ENABLE_FASTBIT 1)
 endif()
 
+#-----------------------------------------------------------------------------
+# Metadata with RocksDB
+#-----------------------------------------------------------------------------
+option(PDC_ENABLE_ROCKSDB "Enable RocksDB." OFF)
+if(PDC_ENABLE_ROCKSDB)
```
Comment (jeanbez) [inline CMakeLists.txt:474]: Could we add a check for a miminum or required version of RocksDB? I'm assuming PDC will not work with any random version of it, so maybe we could add that as a check before enabling it to make it more robust?

**Comment by jeanbez** [comment]
Can you also please add/update the documentation accordingly?

**Review by zhangwei217245 — CHANGES_REQUESTED** [review]
I have created a ticket for this PR: https://github.com/hpc-io/pdc/issues/148

Please take the following changes:

1. RocksDB implementation of metadata and relative metadata search functionality
2. Runtime option for metadata & metadata search functions to call different implementations (passed by RPC parameters)
3. CMake support for enabling/disabling RocksDB . If runtime flag is calling a disabled support, just print do not support and return empty result.
4. Documentation on How metadata is stored in RocksDB and how metadata search is performed, including: where the db file is stored: tmpfs, how metadata tag is stored: rock_key = ObjID_attrKey rock_value=attrVal. How metadata search is done: a full scan across all key-value records in RocksDB.

File: CMakeLists.txt (hunk 1 above)
Comment (zhangwei217245) [reply CMakeLists.txt:474]: We have other dependencies and we do not perform minimum version check for those. But we document the tested version of dependencies. So, putting a few lines of instruction in the installation documentation about the RocksDB version and the commands and options for installing RocksDB should be more effective and efficient than writing a trivial and complex CMake find_package script.

Comment (houjun) [reply CMakeLists.txt:474]: Added min version requirement

**Comment by houjun** [comment]
Added documentation.

**Comment by houjun** [comment]
Going to create a new pull request with both RocksDB and SQLite support.

**Pull request closed without merging**